includes/Modules/Analytics.php

```diff
@@ -1220,21 +1220,40 @@ function ( Google_Service_Analytics_Account $account ) {
 					if ( in_array( $account_id, $account_ids, true ) ) {
 						$properties_profiles = $this->get_data( 'properties-profiles', array( 'accountID' => $account_id ) );
 					} else {
-						// Iterate over each account in reverse so if there is no match,
-						// the last $properties_profiles will be from the first account (selected by default).
-						foreach ( array_reverse( $accounts ) as $account ) {
+						$fallback_properties_profiles = null;
+						$accounts_array               = array_values( $accounts );
+
+						for ( $i = 0, $len = min( 25, count( $accounts_array ) ); $i < $len; $i++ ) {
```

Comment:
Does this need to be declared inside the `for` declaration? It's a bit harder to read this way and I'm not clear on why it needs to be in there.

(Also, `$length` or `$number_of_accounts` might be a more descriptive variable name.)

Reply:
Updated.